addSharedNode

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -e
+
+# Global variables ########################################
+###########################################################
+readonly IFS=$'\n\t'
+readonly ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly CONFIG_DIR="/etc/shippable"
+readonly RUNTIME_DIR="/var/lib/shippable"
+readonly MIGRATIONS_DIR="$ROOT_DIR/migrations"
+readonly POST_INSTALL_MIGRATIONS_DIR="$MIGRATIONS_DIR/post_install"
+readonly SCRIPTS_DIR="$ROOT_DIR/common/scripts"
+readonly LIB_DIR="$SCRIPTS_DIR/lib"
+readonly LOGS_DIR="$RUNTIME_DIR/logs"
+readonly TIMESTAMP="$(date +%Y%m%d%H%M)"
+readonly LOG_FILE="$LOGS_DIR/${TIMESTAMP}_cli.txt"
+readonly LOCAL_SCRIPTS_DIR="$SCRIPTS_DIR/local"
+readonly ADMIRAL_ENV="$CONFIG_DIR/admiral.env"
+readonly SSH_PRIVATE_KEY=$CONFIG_DIR/machinekey
+readonly SSH_PUBLIC_KEY=$CONFIG_DIR/machinekey.pub
+readonly SCRIPTS_DIR_REMOTE="/tmp/shippable"
+readonly MAX_DEFAULT_LOG_COUNT=6
+readonly API_TIMEOUT=600
+readonly ADMIRAL_PORT=50003
+export LC_ALL=C
+export is_db_initialized=false
+export is_default_system_cluster_available=false
+export is_default_system_node_available=false
+export default_runtime_template_id=null
+export archType=null
+export osType=null
+export cluster_type_code=null
+export default_system_cluster_id=null
+
+source "$LIB_DIR/_admiralAdapter.sh"
+source "$LIB_DIR/_logger.sh"
+source "$LIB_DIR/_helpers.sh"
+source $ADMIRAL_ENV
+
+__check_dbStatus() {
+  __process_marker "Checking db status"
+  _shippable_get_systemSettings
+  is_db_initialized=$(echo $response \
+    | jq -r '.db' \
+    | jq -r '.isInitialized')
+
+  __process_msg "is_db_initialized: $is_db_initialized"
+  __process_msg "db status check completed successfully"
+}
+
+__get_systemClusters() {
+  __process_marker "getting systemClusters"
+  _shippable_get_systemClusters 'isDefault=true'
+  if [ "$(echo $response | jq -r '.[0]')" != null ]; then
+    is_default_system_cluster_available=true
+    default_system_cluster_id=$(echo $response | jq -r '.[0].id')
+  fi
+
+   __process_msg "is_default_system_cluster_available: $is_default_system_cluster_available"
+   __process_msg "systemClusters get completed successfully"
+}
+
+
+__get_systemCodes() {
+  __process_marker "getting systemCodes"
+  _shippable_get_systemCodes
+  if [ "$response" != "" ]; then
+  	systemCodes=$response
+    archType=$(echo $response \
+    | jq --arg ARCHITECTURE "$ARCHITECTURE" '.[]
```

Comment:
keep indentation consistent with previous usage (line 43). one tabshift when command spills over to next line